torch/testing/_asserts.py

```diff
@@ -264,57 +318,6 @@ def _check_values_close(
     return AssertionError(msg)
 
 
-def _check_complex_components_individually(
-    check_tensors: Callable[[Tensor, Tensor], Optional[Exception]]
-) -> Callable[[Tensor, Tensor], Optional[Exception]]:
-    """Decorates real-valued tensor check functions to handle complex components individually.
-
-    If the inputs are not complex, this decorator is a no-op.
-
-    Args:
-        check_tensors (Callable[[Tensor, Tensor], Optional[Exception]]): Tensor check function for real-valued tensors.
-
-    Returns:
-        Optional[Exception]: Return value of :attr:`check_tensors`.
-    """
-
-    @functools.wraps(check_tensors)
-    def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[Exception]:
-        if "equal_nan" in kwargs:
-            if kwargs["equal_nan"] == "relaxed":
-                relaxed_complex_nan = True
-                kwargs["equal_nan"] = True
-            else:
-                relaxed_complex_nan = False
-                kwargs["equal_nan"] = bool(kwargs["equal_nan"])
-        else:
-            relaxed_complex_nan = False
-
-        if actual.dtype not in (torch.complex32, torch.complex64, torch.complex128):
-            return check_tensors(actual, expected, **kwargs)  # type: ignore[call-arg]
-
-        if relaxed_complex_nan:
-            actual, expected = [
-                t.clone().masked_fill(
-                    t.real.isnan() | t.imag.isnan(), complex(float("NaN"), float("NaN"))  # type: ignore[call-overload]
-                )
-                for t in (actual, expected)
-            ]
-
-        exc = check_tensors(actual.real, expected.real, **kwargs)  # type: ignore[call-arg]
-        if exc:
-            return _amend_error_message(exc, "{}\n\nThe failure occurred for the real part.")
-
-        exc = check_tensors(actual.imag, expected.imag, **kwargs)  # type: ignore[call-arg]
-        if exc:
-            return _amend_error_message(exc, "{}\n\nThe failure occurred for the imaginary part.")
-
-        return None
-
-    return wrapper
-
-
-@_check_complex_components_individually
```

Comment:
The decorator was misplaced here, since the attribute checking / equalizing should have already happened. This only happens inside `_check_tensors_(equal|close)`. Thus, we now decorate `_check_values_(equal|close)`.

This was only detected now, because we had no test that checked against one complex and one real-valued tensor while `check_dtype` was set to `False`.